sklearn/neighbors/base.py

```diff
@@ -379,11 +379,20 @@ def kneighbors_graph(self, X, n_neighbors=None,
         --------
         NearestNeighbors.radius_neighbors_graph
         """
-        X = check_array(X, accept_sparse=['csr', 'csc', 'coo'])
-
         if n_neighbors is None:
             n_neighbors = self.n_neighbors
 
+        if X is None:
+            new_graph = False
```

Comment:
I didn't mean you could completely factor out the mode case, but you could avoid doing sub-conditions on `new_graph`  (which, I might add, is a confusing variable name).